src/ut_kvp.c

```diff
@@ -366,6 +366,31 @@ uint64_t ut_kvp_getUInt64Field( ut_kvp_instance_t *pInstance, const char *pszKey
     return u64Value;
 }
 
+ut_kvp_status_t ut_kvp_getFormattedFloatField( ut_kvp_instance_t *pInstance, const char *pszKey, char *pszReturnedString, uint32_t uStringSize )
```

Comment:
float ut_kvp_getFloat( )

asset ( float)5.1 == (float)5.100000 )

Reply:
On any error the float / double function should return 0 for all errors.

```c

float stringToFloat(const char *str) {
    char *endptr;
    float value = strtof(str, &endptr);

    // Error checking
    if (*endptr != '\0') {
        fprintf(stderr, "Error: Invalid floating-point string: '%s'\n", str);
        exit(EXIT_FAILURE);  // Or return a special value (e.g., NAN)
    }
    
    return value;
}
```

Support for double, test should be extended for checking double & zero condition.

```c
#include <stdio.h>
#include <stdlib.h>

double stringToDouble(const char *str) {
    char *endptr;
    double value = strtod(str, &endptr);

    // Error checking
    if (*endptr != '\0') {
        fprintf(stderr, "Error: Invalid floating-point string: '%s'\n", str);
        exit(EXIT_FAILURE);  // Or return a special value (e.g., NAN)
    }

    return value;
}

// Example usage
int main() {
    char *str1 = "3.14159265358979323846";
    char *str2 = "-4.2e8";  // scientific notation
    char *str3 = "invalid input"; // error case

    printf("%.15lf\n", stringToDouble(str1));  // Output: 3.141592653589793
    printf("%lf\n", stringToDouble(str2));     // Output: -420000000.000000
    stringToDouble(str3);                     // Prints error message and exits
    return 0;
}
```